src/PrestaShopBundle/Form/Admin/Sell/CartRule/DiscountType.php

```diff
@@ -0,0 +1,129 @@
+use PrestaShop\PrestaShop\Core\Form\ChoiceProvider\DiscountApplicationChoiceProvider;
+use PrestaShopBundle\Form\Admin\Sell\CartRule\EventListener\DiscountListener;
+use PrestaShopBundle\Form\Admin\Type\EntitySearchInputType;
+use PrestaShopBundle\Form\Admin\Type\PriceReductionType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class DiscountType extends TranslatorAwareType
+{
+    /**
+     * @var DiscountListener
+     */
+    private $discountListener;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $languageIsoCode;
+
+    /**
+     * @var DiscountApplicationChoiceProvider
+     */
+    private $discountApplicationChoiceProvider;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        DiscountListener $discountListener,
+        RouterInterface $router,
+        string $employeeIsoCode,
+        DiscountApplicationChoiceProvider $discountApplicationChoiceProvider
+    ) {
+        parent::__construct($translator, $locales);
+        $this->discountListener = $discountListener;
+        $this->router = $router;
+        $this->languageIsoCode = $employeeIsoCode;
+        $this->discountApplicationChoiceProvider = $discountApplicationChoiceProvider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('reduction', PriceReductionType::class, [
+                'currency_select' => true,
+                'label' => false,
+            ])
+            ->add('discount_application', ChoiceType::class, [
+                // choices depends on reduction type data therefore are set in an event subscriber added bellow
+                'choices' => [],
+                'attr' => [
+                    // these attributes are needed for js to change choices when reduction type changes
+                    'data-amount-choices' => json_encode($this->getChoicesByType(Reduction::TYPE_AMOUNT)),
+                    'data-percentage-choices' => json_encode($this->getChoicesByType(Reduction::TYPE_PERCENTAGE)),
+                ],
+            ])
+            ->add('specific_product', EntitySearchInputType::class, [
+                'row_attr' => [
+                    'class' => 'specific-product-search-container',
+                ],
+                'required' => false,
+                'limit' => 1,
+                'min_length' => 3,
+                'label' => $this->trans('Search for specific product', 'Admin.Catalog.Feature'),
+                'remote_url' => $this->router->generate('admin_products_v2_search_associations', [
```

Comment:
ok tried this, seems to work. Reused in Cart rule form everywhere and pack and related product search in product page (but didn't bother about SEO redirect target, because its mixed type of category and product search)